python/ray/util/placement_group.py

```diff
@@ -113,7 +118,12 @@ def _get_none_zero_resource(self, bundle: List[Dict]):
         INFEASIBLE_TASK_SUPPRESS_MAGIC_NUMBER = 0.0101
         for key, value in bundle.items():
             if value > 0:
-                value = INFEASIBLE_TASK_SUPPRESS_MAGIC_NUMBER
+                if key == "memory":
+                    # Make sure the memory resource can be transformed to memory unit.
+                    to_memory_units(value, True)
```

Comment:
I don't understand why this is needed. Isn't this function only for `ready`? If so, the goal of this method is to set the `INFEASIBLE_TASK_SUPPRESS_MAGIC_NUMBER` to avoid warning. Why do we need to handle memory differently?

Reply:
There are two reasons why we need this change:
1. when we set the `memory` resource for the dummy task `bundle_reservation_check`, we can't directly put the `memory` into the `resource`, or it will throw a Runtime exception `The resources dictionary must not contain the key 'memory' or 'object_store_memory'`, you can check this in the method `resources_from_resource_arguments` of python/ray/utils.py
2. The value of `INFEASIBLE_TASK_SUPPRESS_MAGIC_NUMBER` is 0.001, It's ok for CPU and GPU resource to request a fragment resource `0.001`, but for memory, it will request resource by `unit`, I remember the original implements was 50M as a `unit`

Reply:
I see. Can you actually extend the test `test_ready_warning_suppressed` to check memory warning is suppressed?

Reply:
hmmm,  sorry,  I didn't get it, I've checked the method `test_ready_warning_suppressed `,  there seems to be no correlation between them, the `test_ready_warning_suppressed` is used to check whether the infeasible task error msg is suppressed, but in this case, if specified memory resource is less than `MEMORY_RESOURCE_UNIT_BYTES`, it will throw a runtime exception immediately when users invoke `pg.ready()`,  it's a static pre-check.

Reply:
This method is used for `ready`, and the value is supposed to be used to suppress infeasible warning. Please check https://github.com/ray-project/ray/pull/11034

Reply:
Oh, I got it, Thx!

Reply:
I've updated the code, but it seems a little bit problematic here,  the logic of the PUBSUB infeasible task error msg has been changed, it will suppress the error msg as long as the task is related to a placement group bundle, you can check this in the method `PublishInfeasibleTaskError` in `node_manager.cc`,  I think the original design should be suppressed the error msg only for `Pg.ready()` right?

Reply:
hi, @wuisawesome , Can you take a look at this, I think the `INFEASIBLE_TASK_SUPPRESS_MAGIC_NUMBER ` can be removed because the raylet will suppress the error msg as long as the task is related to a placement group bundle in current implements.

Reply:
Yeah I removed the magic number from the raylet. This update looks fine.

Reply:
Got it,  Thx! @wuisawesome